dirvish_1_3_khl/dirvish-runall.8
.\" Automatically generated by Pod::Man v1.34, Pod::Parser v1.13
.\"
.\" Standard preamble:
.\" ========================================================================
.de Sh \" Subsection heading
.br
.if t .Sp
.ne 5
.PP
\fB\\$1\fR
.PP
..
.de Sp \" Vertical space (when we can't use .PP)
.if t .sp .5v
.if n .sp
..
.de Vb \" Begin verbatim text
.ft CW
.nf
.ne \\$1
..
.de Ve \" End verbatim text
.ft R
.fi
..
.\" Set up some character translations and predefined strings.  \*(-- will
.\" give an unbreakable dash, \*(PI will give pi, \*(L" will give a left
.\" double quote, and \*(R" will give a right double quote.  | will give a
.\" real vertical bar.  \*(C+ will give a nicer C++.  Capital omega is used to
.\" do unbreakable dashes and therefore won't be available.  \*(C` and \*(C'
.\" expand to `' in nroff, nothing in troff, for use with C<>.
.tr \(*W-|\(bv\*(Tr
.ds C+ C\v'-.1v'\h'-1p'\s-2+\h'-1p'+\s0\v'.1v'\h'-1p'
.ie n \{\
.    ds -- \(*W-
.    ds PI pi
.    if (\n(.H=4u)&(1m=24u) .ds -- \(*W\h'-12u'\(*W\h'-12u'-\" diablo 10 pitch
.    if (\n(.H=4u)&(1m=20u) .ds -- \(*W\h'-12u'\(*W\h'-8u'-\"  diablo 12 pitch
.    ds L" ""
.    ds R" ""
.    ds C` ""
.    ds C' ""
'br\}
.el\{\
.    ds -- \|\(em\|
.    ds PI \(*p
.    ds L" ``
.    ds R" ''
'br\}
.\"
.\" If the F register is turned on, we'll generate index entries on stderr for
.\" titles (.TH), headers (.SH), subsections (.Sh), items (.Ip), and index
.\" entries marked with X<> in POD.  Of course, you'll have to process the
.\" output yourself in some meaningful fashion.
.if \nF \{\
.    de IX
.    tm Index:\\$1\t\\n%\t"\\$2"
..
.    nr % 0
.    rr F
.\}
.\"
.\" For nroff, turn off justification.  Always turn off hyphenation; it makes
.\" way too many mistakes in technical documents.
.hy 0
.if n .na
.\"
.\" Accent mark definitions (@(#)ms.acc 1.5 88/02/08 SMI; from UCB 4.2).
.\" Fear.  Run.  Save yourself.  No user-serviceable parts.
.    \" fudge factors for nroff and troff
.if n \{\
.    ds #H 0
.    ds #V .8m
.    ds #F .3m
.    ds #[ \f1
.    ds #] \fP
.\}
.if t \{\
.    ds #H ((1u-(\\\\n(.fu%2u))*.13m)
.    ds #V .6m
.    ds #F 0
.    ds #[ \&
.    ds #] \&
.\}
.    \" simple accents for nroff and troff
.if n \{\
.    ds ' \&
.    ds ` \&
.    ds ^ \&
.    ds , \&
.    ds ~ ~
.    ds /
.\}
.if t \{\
.    ds ' \\k:\h'-(\\n(.wu*8/10-\*(#H)'\'\h"|\\n:u"
.    ds ` \\k:\h'-(\\n(.wu*8/10-\*(#H)'\`\h'|\\n:u'
.    ds ^ \\k:\h'-(\\n(.wu*10/11-\*(#H)'^\h'|\\n:u'
.    ds , \\k:\h'-(\\n(.wu*8/10)',\h'|\\n:u'
.    ds ~ \\k:\h'-(\\n(.wu-\*(#H-.1m)'~\h'|\\n:u'
.    ds / \\k:\h'-(\\n(.wu*8/10-\*(#H)'\z\(sl\h'|\\n:u'
.\}
.    \" troff and (daisy-wheel) nroff accents
.ds : \\k:\h'-(\\n(.wu*8/10-\*(#H+.1m+\*(#F)'\v'-\*(#V'\z.\h'.2m+\*(#F'.\h'|\\n:u'\v'\*(#V'
.ds 8 \h'\*(#H'\(*b\h'-\*(#H'
.ds o \\k:\h'-(\\n(.wu+\w'\(de'u-\*(#H)/2u'\v'-.3n'\*(#[\z\(de\v'.3n'\h'|\\n:u'\*(#]
.ds d- \h'\*(#H'\(pd\h'-\w'~'u'\v'-.25m'\f2\(hy\fP\v'.25m'\h'-\*(#H'
.ds D- D\\k:\h'-\w'D'u'\v'-.11m'\z\(hy\v'.11m'\h'|\\n:u'
.ds th \*(#[\v'.3m'\s+1I\s-1\v'-.3m'\h'-(\w'I'u*2/3)'\s-1o\s+1\*(#]
.ds Th \*(#[\s+2I\s-2\h'-\w'I'u*3/5'\v'-.3m'o\v'.3m'\*(#]
.ds ae a\h'-(\w'a'u*4/10)'e
.ds Ae A\h'-(\w'A'u*4/10)'E
.    \" corrections for vroff
.if v .ds ~ \\k:\h'-(\\n(.wu*9/10-\*(#H)'\s-2\u~\d\s+2\h'|\\n:u'
.if v .ds ^ \\k:\h'-(\\n(.wu*10/11-\*(#H)'\v'-.4m'^\v'.4m'\h'|\\n:u'
.    \" for low resolution devices (crt and lpr)
.if \n(.H>23 .if \n(.V>19 \
\{\
.    ds : e
.    ds 8 ss
.    ds o a
.    ds d- d\h'-1'\(ga
.    ds D- D\h'-1'\(hy
.    ds th \o'bp'
.    ds Th \o'LP'
.    ds ae ae
.    ds Ae AE
.\}
.rm #[ #] #H #V #F C
.\" ========================================================================
.\"
.IX Title "DIRVISH-RUNALL 8"
.TH DIRVISH-RUNALL 8 "2005-02-10" "perl v5.8.0" ""
.SH "NAME"
dirvish\-runall \- run a set of dirvish backup jobs.
.SH "SYNOPSIS"
.IX Header "SYNOPSIS"
\&\fBdirvish-runall\fR\fI[\s-1OPTIONS\s0]\fR
.SH "DESCRIPTION"
.IX Header "DESCRIPTION"
Run all the jobs specified in the \fBRunall\fR field of the dirvish master
configuration file.
.SH "OPTIONS"
.IX Header "OPTIONS"
Each option may be unambiguously abbreviated.
.PP
\&\fB\-\-config\fR \fIconfigfile\fR
.PP
Read \fIconfigfile\fR for the \fBRunall:\fR field. By default
\&\fBdirvish-runall\fR will use the \fBRunall:\fR field from the master
configuration file.
.PP
This will only be used by \fBdirvish-runall\fR itself and not be
propagated to the \fBdirvish\fR processes.
.PP
See \fB\f(BIdirvish.conf\fB\|(5)\fR for more details.
.PP
\&\fB\-\-no\-run\fR
.PP
\&\fB\-\-dry\-run\fR
.PP
Don't actually do anything. Just display what would have happened.
.PP
\&\fB\-\-quiet\fR
.PP
Run quietly, only report errors.
.PP
Normally \fBdirvish-runall\fR will report the dirvish commands invoked
with timestamps.
.PP
\&\fB\-\-version\fR
.PP
Print the version information and exit.
.SH "EXIT CODES"
.IX Header "EXIT CODES"
To facilitate further automation and integration of \fBdirvish-runall\fR
with other tools \fBdirvish-runall\fR provides rationalised exit codes.
The exit codes are range based. While the code for a specific error may
change from one version to another it will remain within the specified
range. So don't test for specific exit codes but instead test for a
range of values. To the degree possible higher value ranges indicate
more severe errors.
.IP "\fB0\fR" 10
.IX Item "0"
success
.IP "\fB1\-199\fR" 10
.IX Item "1-199"
Number of \fBdirvish\fR jobs that failed. If more than 199 jobs failed 199
will be the exit code.
.IP "\fB200\-219\fR" 10
.IX Item "200-219"
An error was encountered in loading a configuration file.
.IP "\fB220\-254\fR" 10
.IX Item "220-254"
An error was detected in the configuration.
.IP "\fB255\fR" 10
.IX Item "255"
Incorrect usage.
.SH "FILES"
.IX Header "FILES"
\&\fB/etc/dirvish/master.conf\fR
.PP
alternate master configuration file.
.PP
\&\fB/etc/dirvish.conf\fR
.PP
master configuration file.
.SH "SEE ALSO"
.IX Header "SEE ALSO"
.Vb 1
\& dirvish.conf(5)
.Ve
.SH "AUTHOR"
.IX Header "AUTHOR"
Dirvish was created by J.W. Schultz of Pegasystems Technologies.
.PP
Dirvish is now maintained by Keith Lofstrom at www.dirvish.org , with
the able assistance of many others.
.SH "BUGS"
.IX Header "BUGS"


dirvish_1_2/CHANGELOG
2004/02/25           dirvish.conf.5       12.0       Dirvish-1.2        
                     dirvish.pl           12.0       Dirvish-1.2        
                     install.sh           12.0       Dirvish-1.2        
                     loadconfig.pl        12.0       Dirvish-1.2        
                     RELEASE.html         12.0       Dirvish-1.2        
                     TODO.html            12.0       Dirvish-1.2        
                     dirvish-expire.8     12.0       Dirvish-1.2        
                     dirvish-expire.pl    12.0       Dirvish-1.2        
                     dirvish-locate.8     12.0       Dirvish-1.2        
                     dirvish-locate.pl    12.0       Dirvish-1.2        
                     dirvish-runall.8     12.0       Dirvish-1.2        
                     dirvish-runall.pl    12.0       Dirvish-1.2        
                     dirvish.8            12.0       Dirvish-1.2        
                     COPYING              12.0       Dirvish-1.2        
                     FAQ.html             12.0       Dirvish-1.2        
                     INSTALL              12.0       Dirvish-1.2        
                     RELEASE.html         11.7                         
                     FAQ.html             11.8                         
Preparing to tag 1.2.0

Note license change and prep for 1.2

Update license note.

2004/02/23           dirvish.pl           11.27                        
                     install.sh           11.3                         
Set --delete on rsync so retries will delete deleted files.
					(Thanks to Nathan Hunsperger)

Allow installs to existing install dir with missing subdirs.

2004/01/20           FAQ.html             11.7                         
Removed banner reference.

2004/01/19           FAQ.html             11.6                         
                     dirvish.pl           11.26                        
Reordered chmod operations on image.
					thanks to Nathan Hunsperger

2003/12/21           dirvish.pl           11.25                        
Better handling of vanished files on sender.

2003/12/17           TODO.html            11.16                        
file security check added to wish list

2003/12/15           dirvish.pl           11.24                        
Added recognition for vanishing (unexpected EOF) files on
client.  This happens due to a file being deleted or
truncated between file scan and send.

2003/12/11           FAQ.html             11.5                         
                                          11.4                         
Fixed html tag typo				-- Frank de Jong

2003/12/08           dirvish.pl           11.23                        
Fix permissions of rsync-error file.
					Thanks to Nathan Hunsperger

2003/12/06           INSTALL              11.4                         
Change to reflect rsync's 2.5.7 security release and plan
for post 2.5.6 CVS HEAD to be 2.6.0

2003/12/03           dirvish.conf.5       11.12                        
fixed some references to expire-rules (wrong plural) and
expire_rules (underscore)

2003/11/26           dirvish.conf.5       11.11                        
                     dirvish.pl           11.22                        
                                          11.21                        
                     TODO.html            11.15                        
Added a devices option (default yes) to allow excluding
devices from backups.

Fixed problems with expire-default and expire rules that
assume never.

2003/10/25           dirvish-expire.pl    11.12                        
Allow a vault named dirvish.
A vault named "tree" will still be skipped.

2003/09/17           dirvish.conf.5       11.10                        
fix typo

2003/09/14           dirvish.pl           11.20                        
                     loadconfig.pl        11.13                        
                     TODO.html            11.14                        
Check to be sure that pre and post scripts are specified
correctly (single value) in config file.

Added whitespace after seppuku_prefix

2003/09/01           TODO.html            11.13                        
                     dirvish-expire.pl    11.11                        
fixed dirvish-expire --vault so it works

Still need major rework.

2003/08/30           TODO.html            11.12                        
2003/08/29           dirvish.pl           11.19                        
                                          11.18                        
                     loadconfig.pl        11.12                        
                     TODO.html            11.11                        
                     INSTALL              11.3                         
                     FAQ.html             11.3                         
                     dirvish.conf.5       11.9                         
Fix seppuku prefix so it is $vault:$branch:$image

Added seppuku prefix variable so error messages will be
suitably marked with the vault:

Better reflect filesystem requirements in INSTALL and FAQ.

2003/08/03           dirvish-expire.pl    11.10                        
                     dirvish-locate.pl    11.9                         
                     dirvish-runall.pl    11.9                         
                     dirvish.pl           11.17                        
                     loadconfig.pl        11.11                        
                     COPYING              11.5                         
                                          11.4                         
                     dirvish-expire.pl    11.9                         
                     dirvish-locate.pl    11.8                         
                     dirvish-runall.pl    11.8                         
                     dirvish.pl           11.16                        
                     loadconfig.pl        11.10                        

dirvish_1_2_1/dirvish-runall.pl

#       $Id: dirvish-runall.pl,v 12.0 2004/02/25 02:42:14 jw Exp $  $Name: Dirvish-1_2 $

$VERSION = ('$Name: Dirvish-1_2_1 $' =~ /Dirvish/i)
	? ('$Name: Dirvish-1_2_1 $' =~ m/^.*:\s+dirvish-(.*)\s*\$$/i)[0]
	: '1.1.2 patch' . ('$Id: dirvish-runall.pl,v 12.0 2004/02/25 02:42:14 jw Exp $'
		=~ m/^.*,v(.*:\d\d)\s.*$/)[0];
$VERSION =~ s/_/./g;


#########################################################################
#                                                         		#
#	Copyright 2002 and $Date: 2004/02/25 02:42:14 $
#                         Pegasystems Technologies and J.W. Schultz 	#
#                                                         		#
#	Licensed under the Open Software License version 2.0		#
#                                                         		#
#	This program is free software; you can redistribute it		#
#	and/or modify it under the terms of the Open Software		#
#	License, version 2.0 by Lauwrence E. Rosen.			#
#                                                         		#
#	This program is distributed in the hope that it will be		#
#	useful, but WITHOUT ANY WARRANTY; without even the implied	#
#	warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR		#
#	PURPOSE.  See the Open Software License for details.		#
#                                                         		#
#########################################################################

use Time::ParseDate;
use POSIX qw(strftime);
use Getopt::Long;

sub loadconfig;
sub seppuku;

sub usage
{
	my $message = shift(@_);

	length($message) and print STDERR $message, "\n\n";

	$! and exit(255); # because getopt seems to send us here for death

	print STDERR <<EOUSAGE;
USAGE
	dirvish-runall OPTIONS
OPTIONS
	--config configfile
	--no-run
	--quiet
	--version
EOUSAGE

	exit 255;
}

$Options = {
	version		=> sub {
			print STDERR "dirvish version $VERSION\n";
			exit(0);
		},
	help		=> \&usage,
};

GetOptions($Options, qw(
		config=s
		quiet no-run|dry-run
		version help
	)) or usage;

if ($$Options{config})
{
	$Config = loadconfig(undef, $$Options{config})
}
elsif ($CONFDIR =~ /dirvish$/ && -f "$CONFDIR.conf")
{
	$Config = loadconfig(undef, "$CONFDIR.conf");
}
elsif (-f "$CONFDIR/master.conf")
{
	$Config = loadconfig(undef, "$CONFDIR/master.conf");
}
elsif (-f "$CONFDIR/dirvish.conf")
{
	seppuku 250, <<EOERR;
ERROR: no master configuration file.
	An old $CONFDIR/dirvish.conf file found.
	Please read the dirvish release notes.
EOERR
}
else
{
	seppuku 251, "ERROR: no global configuration file";
}
$$Config{Dirvish} ||= 'dirvish';

$$Options{'no-run'} and $$Options{quiet} = 0;

$errors = 0;

for $sched (@{$$Config{Runall}})
{
	($vault, $itime) = split(/\s+/, $sched);
	$cmd = "$$Config{Dirvish} --vault $vault";
	$itime and $cmd .= qq[ --image-time "$itime"];
	$$Options{quiet}
		or printf "%s %s\n", strftime('%H:%M:%S', localtime), $cmd;
	$$Options{'no-run'} and next;
	$status = system($cmd);
	$status < 0 || $status / 256 and ++$errors;

}
$$Options{quiet}
	or printf "%s %s\n", strftime('%H:%M:%S', localtime), 'done';


exit ($errors < 200 ? $errors : 199);
